packages/batch/tests/helpers/handlers.ts

```diff
@@ -8,50 +9,70 @@ import type {
 const sqsRecordHandler = (record: SQSRecord): string => {
   const body = record.body;
   if (body.includes('fail')) {
-    throw Error('Failed to process record.');
+    throw new Error('Failed to process record.');
   }
 
   return body;
 };
 
-const asyncSqsRecordHandler = async (record: SQSRecord): Promise<string> =>
-  Promise.resolve(sqsRecordHandler(record));
+const asyncSqsRecordHandler = async (record: SQSRecord): Promise<string> => {
+  const body = record.body;
+  if (body.includes('fail')) {
+    throw new Error('Failed to process record.');
+  }
+  await new Promise((resolve) => setTimeout(resolve, 1));
+  return body;
+};
 
 const kinesisRecordHandler = (record: KinesisStreamRecord): string => {
   const body = record.kinesis.data;
   if (body.includes('fail')) {
-    throw Error('Failed to process record.');
+    throw new Error('Failed to process record.');
   }
 
   return body;
 };
 
 const asyncKinesisRecordHandler = async (
   record: KinesisStreamRecord
-): Promise<string> => Promise.resolve(kinesisRecordHandler(record));
+): Promise<string> => {
+  const body = record.kinesis.data;
+  if (body.includes('fail')) {
+    throw new Error('Failed to process record.');
+  }
+  await new Promise((resolve) => setTimeout(resolve, 1));
```

Comment:
You can use the built in promise version of `setTimeout` rather than manually using the promise constuctor:

```ts
import {setTimeout} from 'node:timers/promises;

// ...

await setTimeout(1);
```

Reply:
Nice, didn't know this existed - changing it.